test/python/taskbasedsimulator/benchmark.py

```diff
@@ -1,148 +1,422 @@
+    p = Path(__file__).with_name('results_verification.csv')
+    with p.open('a+') as file:
         file.write(';{};{};{};{};{};{}\n'.format(qc.name, qc.num_qubits, qc.size(), mode, run_results['time_setup'],
                                                  run_results['time_sim']))
-
     print(qc.name, qc.num_qubits, qc.size(), mode, run_results['time_setup'], run_results['time_sim'],
           sep=';')
 
 
-def execute_verification_all(qc, backend, basis_gates, shots, include_cotengra, max_time, max_repeats, plot_ring):
-    print('Transpiling circuit')
-    qc = transpile(qc, basis_gates=basis_gates, optimization_level=0)
-    execute_verification(qc, backend, shots)
-    execute_verification(qc, backend, shots, 'alternating')
+def execute_verification_all(qc, qcog, gatecosts, backend, shots, include_cotengra, max_time, max_repeats, plot_ring):
+    print('Running sequential')
+    execute_verification(qc, qcog, gatecosts, backend, shots)
+    #print('Running naive')
+    #execute_verification(qc, qcog, gatecosts, backend, shots, 'alternating')
+    print('Running gatecost')
+    execute_verification(qc, qcog, gatecosts, backend, shots, 'gatecost')
+    print('Running CoTenGra')
     if include_cotengra:
-        execute_verification(qc, backend, shots, 'cotengra', cotengra_max_time=max_time,
+        execute_verification(qc, qcog, gatecosts, backend, shots, 'cotengra', cotengra_max_time=max_time,
                              cotengra_max_repeats=max_repeats, cotengra_plot_ring=plot_ring)
 
-
 if __name__ == '__main__':
-    with open('results_verification.csv', 'a+') as file:
+    p = Path(__file__).with_name('results_verification.csv')
+    with p.open('a+') as file:
         file.write(
             datetime.datetime.now().strftime("%Y-%m-%d_%H:%M:%S") + ';name;nqubits;ngates;mode;time_setup;time_sim\n')
 
     # settings
     backend = PathQasmSimulator()
     configuration_dict = backend.configuration().to_dict()
-    basis_gates = configuration_dict['basis_gates']
+    basis_gates = ["id", "rz", "sx", "x", "cx", "reset"]
+    basis_gates_transpile = configuration_dict['basis_gates']
+    # loading the Gatecost LUT
+    script_path = os.path.abspath(__file__)
+    script_dir = os.path.split(script_path)[0]
+    print(script_dir)
+    rel_path = "qiskit_O2_nonancilla.profile"
+    abs_file_path = os.path.join(script_dir, rel_path)
+    lut_gatecost = {}
+    with open(
+            abs_file_path,
+            'r') as f:
+        for line in f.readlines():
+            line = line.split(" ")
+            lut_gatecost[line[0] + line[1]] = str(line[2][:-1])
 
     shots = 0
     max_time = 3600
     max_repeats = 256
     plot_ring = False
-
     # run benchmarks
-    for n in [48, 64, 96, 128]:
-        qc = ghz(n, include_measurements=False)
-        execute_verification_all(qc=qc, backend=backend, basis_gates=basis_gates, shots=shots, include_cotengra=True,
+    for n in [10, 11, 12, 13, 14]:
```

Comment:
starting from here this contains terrible amounts of code duplication.
Please refactor the common functionality into a separate function that is being called.
To this end, consider creating a dictionary that maps names of the benchmarks to the respective qubit numbers that should be evaluated.